File: exercise/python/day02/games/fizz_buzz.py
from typing import Optional

MIN = 1
MAX = 100
FIZZBUZZ = 15
FIZZ = 3
BUZZ = 5


class FizzBuzz:
    @staticmethod
    def convert(input: int) -> Optional[str]:
        if FizzBuzz.is_out_of_range(input):
            return None
        else:
            return FizzBuzz.convert_safely(input)

    @staticmethod
    def convert_safely(input: int) -> str:
        if FizzBuzz.is_divisible_by(FIZZBUZZ, input):
            return "FizzBuzz"
        elif FizzBuzz.is_divisible_by(FIZZ, input):
            return "Fizz"
        elif FizzBuzz.is_divisible_by(BUZZ, input):
            return "Buzz"
        else:
            return str(input)

    @staticmethod
    def is_divisible_by(divisor: int, input: int) -> bool:
        return input % divisor == 0

    @staticmethod
    def is_out_of_range(input: int) -> bool:
        return input < MIN or input > MAX

File: exercise/php/day02/src/Games/FizzBuzz.php
<?php

declare(strict_types = 1);

namespace Games {
	class FizzBuzz
	{
		public const MIN = 1;

		public const MAX = 100;

		private static array $mapping = [
			15 => 'FizzBuzz',
			3 => 'Fizz',
			5 => 'Buzz',
		];

		public static function convert(int $input): ?string
		{
			if (self::isOutOfRange($input)) {
				return null;
			}

			return self::convertSafely($input);
		}

		private static function isOutOfRange(int $input): bool
		{
			return self::MIN > $input || self::MAX < $input;
		}

		private static function convertSafely(int $input): string
		{
			foreach (self::$mapping as $key => $value) {
				if (self::isDivisibleBy($input, $key)) {
					return $value;
				}
			}

			return (string) $input;
		}

		private static function isDivisibleBy(int $input, int $divisor): bool
		{
			return $input % $divisor === 0;
		}
	}
}
